### ISSUE TITLE
number-literal-format fix strips significant zeros

### ISSUE THREAD

**Issue by rgant** [palantir/tslint#4855]
### Bug Report

- __TSLint version__: 5.13.0
- __TypeScript version__: 3.4.5
- __Running TSLint via__: yarn test

#### TypeScript code being linted

```ts
const rate = 0.022000;
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint:recommended"],
    "rules": {
        "number-literal-format": true
    } // add additional rules and their configuration
}
```

#### Actual behavior

```
- const rate = 0.022000;
+ const rate = 22000;
```

#### Expected behavior

```
- const rate = 0.022000;
+ const rate = 0.022;
```

Working branch for future pull request: https://github.com/rgant/tslint/tree/bugfix/number-literal-format-fix

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
